controller/api/api.go

```diff
@@ -612,10 +612,19 @@ func (a *Api) scaleContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := a.manager.ScaleContainer(containerId, numInstances); err != nil {
+	result := a.manager.ScaleContainer(containerId, numInstances)
+	if len(result.Errors) > 0 {
+		http.Error(w, string(json), http.StatusInternalServerError)
+		return
+	}
+
+	json, err := json.Marshal(result)
```

Comment:
There is actually an easier way to do this:

`
if err := json.NewEncoder(w).Encode(result); err != nil {
	http.Error(w, err.String(), http.StatusInternalServerError)
        return
}